File: week1/pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
int nod(long long k);
int first(long long k);
int first2(long long x);
int main(void)
{
    int sump = 0, sumi = 0, d1 = 0;
    long long n=get_long_long("Number: ");
    long long nc=n;
    int l=nod(n);
    for(int i=0;i<l;i++){
        d1 = n % 10;
        if (i % 2 == 0)
            sump += d1;
        else{
            int multiply = 2 * d1;
            sumi += (multiply / 10) + (multiply % 10);
            }
        n /= 10;
     }
    bool checksum = (sump + sumi) % 10 == 0;
    //printf("%d",first2(nc3));

    if (nod(nc) >= 13 && nod(nc) <= 16 && checksum && first(nc)==4){
        printf("VISA\n");
    }
    else if (nod(nc) == 16 && checksum && first2(nc)>=51 && first2(nc)<=55){
        printf("MASTERCARD\n");
    }
    else if (nod(nc) == 15 && checksum && (first2(nc)==34 || first2(nc)==37)){
        printf("AMEX\n");
    }
    else{
        printf("INVALID\n");
    }

}

int nod(long long k){
    int number_of_digits = 0;
    do{
     ++number_of_digits;
     k /= 10;
     }while (k);
    return number_of_digits;
}
int first(long long k){
    while (k >= 10)
    {
        k /= 10;
    }
    return k;
}
int first2(long long x){
    while (x >= 100)
    {
        x /= 10;
    }
    return x;
}

File: week6/credit.py
from cs50 import *

def main():
    strcard = input("Number: ")
    cc_number=int(strcard)
    #print(cc_number)
    #print(len(strcard))
    l = list(strcard)
    digit1 = 0
    digit2 = 0
    sum_of_double_odds = 0
    sum_of_evens = 0
    num_digits = 0;

    while cc_number > 0:
        digit2=digit1
        digit1 = cc_number % 10

        if num_digits % 2 == 0:
            sum_of_evens += digit1
        else:
            multiple = 2 * digit1
            sum_of_double_odds += (multiple // 10) + (multiple % 10)

        cc_number //=10
        num_digits += 1

    checksum = (sum_of_evens + sum_of_double_odds) % 10 == 0
    #print(checksum)


    if len(strcard) >= 13 and len(strcard) <= 16 and l[0] =='4' and checksum:
        print("VISA")
    elif len(strcard) == 16 and l[0]=='5' and (l[1]=='1' or l[1]=='2' or l[1]=='3' or l[2]=='4' or l[1]=='5') and checksum:
        print("MASTERCARD")
    elif len(strcard) == 15 and l[0]=='3' and (l[1]=='4' or l[1]=='7') and checksum:
        print("AMEX")
    else:
        print("INVALID")


if __name__ == "__main__":
    main()